python/ray/tune/integration/kubernetes.py

```diff
@@ -0,0 +1,143 @@
+    """
+
+    class _NamespacedKubernetesSyncer(KubernetesSyncer):
+        _namespace = namespace
+
+    return _NamespacedKubernetesSyncer
+
+
+class KubernetesSyncer(NodeSyncer):
+    """
+    KubernetesSyncer used for synchronization of checkpoints between
+    Kubernetes pods.
+
+    This syncer extends the node syncer, but is usually instantiated
+    without a custom sync client. The sync client defaults to
+    ``KubernetesSyncClient`` instead.
+
+    KubernetesSyncer uses the default namespace ``ray``. You should
+    probably use ``NamespacedKubernetesSyncer`` to return a class
+    with a custom namespace instead.
+    """
+
+    _namespace = "ray"
+
+    def __init__(self, local_dir, remote_dir, sync_client=None):
+        self.local_ip = services.get_node_ip_address()
+        self.local_node = self._get_kubernetes_node_by_ip(self.local_ip)
+        self.worker_ip = None
+        self.worker_node = None
+
+        sync_client = sync_client or KubernetesSyncClient(
+            namespace=self.__class__._namespace)
+
+        super(NodeSyncer, self).__init__(local_dir, remote_dir, sync_client)
+
+    def set_worker_ip(self, worker_ip):
+        self.worker_ip = worker_ip
+        self.worker_node = self._get_kubernetes_node_by_ip(worker_ip)
+
+    def _get_kubernetes_node_by_ip(self, node_ip):
+        """Return node name by internal or external IP"""
+        kubernetes.config.load_incluster_config()
+        api = kubernetes.client.CoreV1Api()
+        pods = api.list_namespaced_pod(self._namespace)
+        for pod in pods.items:
+            if pod.status.host_ip == node_ip or \
+               pod.status.pod_ip == node_ip:
+                return pod.metadata.name
+
+        logger.error(
+            "Could not find Kubernetes pod name for IP {}".format(node_ip))
+        return None
+
+    @property
+    def _remote_path(self):
+        return (self.worker_node, self._remote_dir)
+
+
+class KubernetesSyncClient(SyncClient):
+    """KubernetesSyncClient to be used by KubernetesSyncer.
+
+    This client takes care of executing the synchronization
+    commands for Kubernetes clients. In its ``sync_down`` and
+    ``sync_up`` commands, it expects tuples for the source
+    and target, respectively, for compatibility with the
+    KubernetesCommandRunner.
+
+    Args:
+        namespace (str): Namespace in which the pods live.
+        process_runner: How commands should be called.
+            Defaults to ``subprocess``.
+
+    """
+
+    def __init__(self, namespace, process_runner=subprocess):
+        self.namespace = namespace
+        self._process_runner = process_runner
+        self._command_runners = {}
+
+    def _create_command_runner(self, node_id):
+        """Create a command runner for one Kubernetes node"""
+        return KubernetesCommandRunner(
+            log_prefix="KubernetesSyncClient: {}:".format(node_id),
+            namespace=self.namespace,
+            node_id=node_id,
+            auth_config=None,
```

Comment:
would there be a case where we need the auth_config?

Reply:
`KubernetesCommandRunner` doesn't use the `auth_config`, it is completely ignored. I guess the reason the argument is there is because it implements the `CommandRunnerInterface`. 
However, since it doesn't use a default value we will have to pass `None` explicitly.